paimon-core/src/main/java/org/apache/paimon/manifest/ManifestFileMeta.java

```diff
@@ -185,4 +207,108 @@ private static void mergeCandidates(
             newMetas.addAll(merged);
         }
     }
+
+    static class FullCompactionHelper {
+        private List<ManifestFileMeta> metas;
+        private ManifestFile manifestFile;
+        private RowType partitionType;
+        private int index;
+
+        public FullCompactionHelper(
+                List<ManifestFileMeta> metas, ManifestFile manifestFile, RowType partitionType) {
+            this.metas = metas;
+            this.manifestFile = manifestFile;
+            this.partitionType = partitionType;
+            int index = metas.size();
+        }
+
+        public int getFirstManifest() {
+            // Manifests which container DELETE ManifestEntry
+            List<ManifestFileMeta> delta = Lists.newArrayList();
+            long deltaSize = 0;
```

Comment:
I've thought about the strategy here and perhaps a better one would be:
Collect the delta part that follows, and if there are more deletes in the delta than adds in the delta, start full-compaction.